bpf/bpf_host.c

```diff
@@ -212,7 +212,13 @@ handle_ipv6(struct __ctx_buff *ctx, __u32 secctx, const bool from_host)
 		if (ctx_get_xfer(ctx) != XFER_PKT_NO_SVC &&
 		    !bpf_skip_nodeport(ctx)) {
 			ret = nodeport_lb6(ctx, secctx);
-			if (ret < 0)
+			/* nodeport_lb6() returns with TC_ACT_REDIRECT for
+			 * traffic to L7 LB. Policy enforcement needs to take
+			 * place after L7 LB has processed the packet, so we
+			 * return to stack immediately here with
+			 * TC_ACT_REDIRECT.
+			 */
+			if (ret < 0 || ret == TC_ACT_REDIRECT)
 				return ret;
 		}
```

Comment:
I believe we need the same check in `bpf_overlay.c`'s nodeport code:
https://github.com/cilium/cilium/blob/34d128aa85c61a41175c92b64afacc05bcd615a5/bpf/bpf_overlay.c#L57-L65

Reply:
Was this addressed?